projects/cli-files/go-ls/cmd/root.go

```diff
@@ -0,0 +1,75 @@
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+func Execute() {
+	// define a flag of type boolean with the specific properties (name, value, usage)
+	helpFlag := flag.Bool("h", false, "show help")
+
+	// parse the command line flags
+	flag.Parse()
+
+	// if the "-h" flag is provided
+	if *helpFlag {
+		fmt.Fprintf(os.Stdout, "go-ls help message\n")
+		return
+	}
+
+	// initialise/default the directory path to where go-ls was called from
+	directoryPath, err := os.Getwd()
+
+	// if we can't get the current working directory
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to get the current working directory: %v\n", err)
+		os.Exit(2)
+	}
```

Comment:
Do you always actually need to be able to get the current working directory?

If not, is there a way to avoid erroring when we don't actually need to get it?

Reply:
I have removed this logic and tried to rewrite it without